### DISCUSSION TITLE
Pick fields in API response?

### DISCUSSION THREAD

**Discussion by omaksi** [strapi/strapi#7615] [Legacy Topics]
Hello, I'd like to ask if it's possible to pick/filter fields which are returned in an API response.

example: For a list of records I don't need a long description field in each record, but for the detail I do. How to best solve this? Thanks!

**Comment by derrickmehaffy** [comment]
Currently no that is not a feature without doing something like this: https://strapi.io/documentation/v3.x/guides/custom-data-response.html

Or you create a custom query: https://strapi.io/documentation/v3.x/concepts/queries.html#custom-queries

**Reply by omaksi** [reply]
Thanks, I thought so. 
Should I create a feature suggestion issue for this?

**Reply by derrickmehaffy** [reply]
Please do, and please don't forget to post on the ProductBoard (link is in the Feature Request template) and include a link to the Github issue you made :smile:

**Reply by omaksi** [reply]
Actually it has already been considered here: 

https://portal.productboard.com/strapi/1-roadmap/c/31-exclude-include-fields-using-url-parameters
https://github.com/strapi/strapi/issues/2086
